### ISSUE TITLE
is it possible of BBB_eMMC ?

### ISSUE THREAD

**Issue by michael1888** [ddlawrence/BBB_MMC#1]
Hi ddlawrence,
Thank you for the sharing,
I am seeking NAND boot of beaglebone black with starterware & ccs, which in general requests BBB_eMMC. 
Have you any experience to sharing on BBB_eMMC? Any advice and suggestions will be greatly appreciated ...
Best

**Comment by ddlawrence** [comment]
hi michael.  sorry for the late reply. i had to work this week.no, i
do not have emmc running.  i really want this also, but must continue
with ethernet/ip stuff.  

i advise against using code composter, use gnu and image building from
the command line.  i got it working with windows and linux.  

i can provide you with an old, very simple, limited mmc program.  and
you may be able to adapt it to work with emmc.  it may not be too
difficult.  theoretically, just change the mmc0 base address to point
to mmc1.

if i make some extra cash i will pay you to do it for me.  

keep in touch..........dd  

d7 at hushmail dot com 
ps what is the significance of 1888?

On 3/26/2018 at 2:33 PM, "michael1888"  wrote:

	Hi ddlawrence,
 Thank you for the sharing,
 I am seeking NAND boot of beaglebone black with starterware & ccs,
which in general requests BBB_eMMC.
 Have you any experience to sharing on BBB_eMMC? Any advice and
suggestions will be greatly appreciated ...
 Best

**Comment by Lennart99** [comment]
Sorry for bumbing this thread, but is there any progress made on this?
I really want a to be able to access the eMMC on the BBB from within my bare-metal applications

**Comment by ddlawrence** [comment]
hi lennart.  sorry i did not work on this.  and i have other more urgent issues.  
check back in a month.  i may take a stab at it.  still u will need to migrate to gnu.  
ti does not support starterware, and i gave up on it and ccs a long time ago. 
later...........dd

**Comment by Lennart99** [comment]
I don' t think porting to GNU will be too hard, I already have compiled the code with arm-none-eabi, just have to change the addresses to load the code into internal ram or am I missing something else?

Also: if you could give me some pointers on how to port this for eMMC, I can also try to do it myself

let me know what you think 
Lennart

**Comment by ddlawrence** [comment]
yes, just compile link run the mmc demo application using the given Makefile and memmap.lds
the Makefile has comments on how to do it under Linux or Windows
get it running, then mod it using MMC1 and attempt to only read the filesystem first
if u can read it, tell me right away.
happy hacking...........dd
